### PR TITLE
Add Rules Notifier Block

### PR DESCRIPTION
**Pull request by hobinjk**
I tried to split this into logical commits but it's a bit much even then

### CODE REVIEW

File: src/rules-engine/effects/NotifierOutletEffect.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      console.warn(`Notifier "${this.notifier}" not found, unable to notify`);
+      return;
+    }
+    const outlet = notifier.getOutlet(this.outlet);
+    if (!outlet) {
+      console.warn(`Outlet "${this.outlet}" of notifier "${this.notifier}" not found, unable to notify`);
+      return;
+    }
+
+    outlet.notify(this.title, this.message, this.level);
```
Comment (mrstegeman) [inline src/rules-engine/effects/NotifierOutletEffect.js]: `notify()` returns a `Promise`, so you may want to catch rejections here.

File: static/js/views/rule-screen.js [hunk 2]
```diff
@@ -310,6 +283,11 @@ const RuleScreen = {
       return elt.parentNode;
     }).filter(isValidSelection);
 
+    if (triggerBlocks.length === 0 || effectBlocks.length === 0) {
+      console.log('WHY', triggerBlocks, effectBlocks);
```
Comment (mrstegeman) [inline static/js/views/rule-screen.js]: Haha probably remove the log.

Comment (hobinjk) [reply static/js/views/rule-screen.js]: Ohhh, I saw that during my sweep for console.logs but I thought it was there already, woops

File: src/rules-engine/effects/NotifierOutletEffect.js (hunk 1 above)
Comment (hobinjk) [reply src/rules-engine/effects/NotifierOutletEffect.js]: :+1:

**Comment by mrstegeman** [comment]
Looks better! A couple things:

1. Are you experiencing the "undefined" thing name issue, or is it just me?
2. I noticed this in the log. We may need to delay start-up on the rules engine a bit. The rule does still seem to work fine, though.

```
2019-07-11 11:38:12.712 ERROR  : Error getting value for thingId: virtual-things-0 property: on
2019-07-11 11:38:12.712 ERROR  : getProperty: device: virtual-things-0 not found.
2019-07-11 11:38:12.713 WARN   : Rule get failed { code: 500,
  message: 'getProperty: device: virtual-things-0 not found.' }
2019-07-11 11:38:12.720 INFO   : Loading add-on: virtual-things-adapter
2019-07-11 11:38:12.727 INFO   : Loading add-on: email-sender-adapter
2019-07-11 11:38:12.876 INFO   : virtual-things: Opening database: /Users/mstegeman/.mozilla-iot/config/db.sqlite3
2019-07-11 11:38:12.884 INFO   : email-sender: Opening database: /Users/mstegeman/.mozilla-iot/config/db.sqlite3
2019-07-11 11:38:12.887 INFO   : virtual-things: Loading add-on for virtual-things-adapter from /Users/mstegeman/.mozilla-iot/addons/virtual-things-adapter
2019-07-11 11:38:12.895 INFO   : email-sender: Loading add-on for email-sender-adapter from /Users/mstegeman/.mozilla-iot/addons/email-sender-adapter
```

**Comment by hobinjk** [comment]
I'm also experiencing the undefined thing but none of my things have names on the homescreen either.

That error message is harmless and expected based on the startup process of the rules engine. The getInitialValue function could check for the existence of the thing first but it's not particularly harmful in its current state

**Comment by mrstegeman** [comment]
> I'm also experiencing the undefined thing but none of my things have names on the homescreen either.

Weird. I do have them on the home screen, as well as the blocks after I drag them into place. Just not in the bottom list. I'm guessing for you, you just need to update your `gateway-addon` dependency.

**Pull request merged**